pyznap/send.py

```diff
@@ -64,10 +65,14 @@ def send_snap(snapshot, dest_name, base=None, ssh_dest=None, raw=False):
         send.stderr.close()
 
         stdout, stderr = recv.communicate()
+        if stderr.decode().startswith('cannot receive incremental stream'):
+            raise SSHConnectError(stderr)
         # raise any error that occured
         if recv.returncode:
             raise CalledProcessError(returncode=recv.returncode, cmd=recv.args, output=stdout, stderr=stderr)
 
+    except SSHConnectError as err:
+        raise
```

Comment:
This method should not raise an error it doesn't handle itself, except for the `KeyboardInterrupt`, which is then handled further up. This leads to a crash in pyznap if you do the following for example:

    $ pyznap send -s test0 -d test1 -r
    $ ctrl+C
    $ pyznap send -s test0 -d test1

The second `pyznap send` fails as there is a partial receive token and thus raises the `SSHConnectError`, but this is not handled anywhere in the code, so pyznap crashes.

Reply:
I think we should remove the `SSHConnectError` and maybe add another exception class for `cannot receive incremental stream`, as said above.

Reply:
> This method should not raise an error it doesn't handle itself, except for the `KeyboardInterrupt`, which is then handled further up. This leads to a crash in pyznap if you do the following for example:
> 
> ```
> $ pyznap send -s test0 -d test1 -r
> $ ctrl+C
> $ pyznap send -s test0 -d test1
> ```
> 
> The second `pyznap send` fails as there is a partial receive token and thus raises the `SSHConnectError`, but this is not handled anywhere in the code, so pyznap crashes.

can you fix this bug?

Reply:
Yes, will do.

Reply:
**One important thing**
`74 except SSHConnectError as err:`
this exception is either raised by `send`(in line 68, process.py) or `recv`(in line 68, send.py), I don't know which one will raise the error first, so i catch both.
`send` will raise an 'SSHConnectError' and `recv` will have error msg: `cannot receive incremental stream`
That's the reason I choose `SSHConnectError` as the name here, because `send`(in line 68, process.py) will raise a `SSHConnectError`